rdtset/rdt.c

```diff
@@ -1449,6 +1489,8 @@ cfg_set_cores_os(const unsigned technology, const cpu_set_t *cores,
                 /* Get cores on res id i */
                 if (technology & (1 << PQOS_CAP_TYPE_L2CA))
                         ret = get_l2id_cores(cores, i, &core_num, core_array);
+		else if (technology & (1 << PQOS_CAP_TYPE_MBA))
```

Comment:
All 3 technologies can be used at the same time. It won't work when division into mba/cat is not the same.

Reply:
Updated the functions to handle each technology separately. Tested and verified also.